Help: Add courses button to /help

omarjackman opened the pull request:
This pull request adds a courses button to /help. Right now the button doesn't lead anywhere because the new courses page will be built in a later PR.

![screen shot 2016-08-22 at 12 30 18 pm](https://cloud.githubusercontent.com/assets/1854440/17862709/67982206-6864-11e6-9066-35b10dbdbcf7.png)

#### How to test
 1. Navigate to https://calypso.live/help?branch=add/help-courses-button
 2. Notice the new courses button directly below the help search box.

Test live: https://calypso.live/?branch=add/help-courses-button

dllh commented on client/me/help/main.jsx:
```diff
@@ -73,32 +80,61 @@ const Help = React.createClass( {
+	getCoursesLink: function() {
+		return (
+			<div className="help__course-links">
+				<Card href="#/help/courses">
```
I'm not sure whether the `#` here is a typo or simply a way to indicate what the url will be once it's been implemented, without sending the user to a broken link.

omarjackman replied:
Its just a place holder so we don't go to a broken link

dllh commented on client/me/help/controller.js:
```diff
@@ -25,7 +25,7 @@ module.exports = {
 
 		renderWithReduxStore(
-			React.createElement( Help ),
+			<Help isCoursesEnabled={ config.isEnabled( 'help/courses' ) }/>,
```
Should we go ahead and abstract this out into a function or something, so that it's easy later to plug in more rules (like one to display it only for eligible users)?

omarjackman replied:
I added another prop based on this called `showCourseTeaser` in @cd198cf2803d8e020c05626dd6c011105d4c60a3

Eventually it will be composed with more business logic.

dllh commented on client/me/help/main.jsx:
```diff
@@ -73,32 +80,61 @@ const Help = React.createClass( {
 				<MeSidebarNavigation />
 				<HelpSearch />
-				{ ! this.props.isEmailVerified && <HelpUnverifiedWarning /> }
+				{ isCoursesEnabled && this.getCoursesLink() }
```
Kind of a nit-pick, but I'd expect a `getCoursesLink()` function to return a link, when it really returns an element or component. Maybe `getCoursesTeaser()` or something similar is slightly more descriptive?

omarjackman replied:
I changed the name to `getCoursesTeaser` in @cd198cf2803d8e020c05626dd6c011105d4c60a3

dllh commented on client/me/help/main.jsx:
```diff
@@ -73,32 +80,61 @@ const Help = React.createClass( {
 
+	getCoursesLink: function() {
+		return (
+			<div className="help__course-links">
```
I feel like there should be a component for this, but I'm not seeing one [here](https://wpcalypso.wordpress.com/devdocs/design), so maybe there's not. I'm fine with shipping as is, but if @mrheston can point us to a similar UI that uses a component we can drop in here, we should submit another PR later to use it.

mrheston replied:
There's a very, very similar component here; in fact, this design is 100% based on that:

https://wpcalypso.wordpress.com/devdocs/blocks/domain-tip

I asked @mtias about creating a generic component so we could use it in other places if needed, and he suggested extending the `.card.is-link` component with a `highlighted` property (or `primary`, or another similar word): `<Card highlighted href="">`.

So `<Card highlighted>` should return `<div class="card is-highlighted">` and we could control the styling with the `.is-highlighted` class.

omarjackman replied:
There's some duplication here in look but not really in code since theres no component I could find that was reusable.

dllh commented:
I tested this a bit, and the code seems straightforward. One thing I wasn't 100% sure about was how the props would be passed through (I'm professing ignorance here more than skepticism). Now it looks like we're explicitly passing in `isCoursesEnabled`, and I guess the `isEmailVerified` prop is set or implied somewhere else in the code tree, so that the changes here are ok. I'd maybe want to test with an unverified account to make sure we've introduced no regression here, but if that scenario behaves as expected, I feel good enough about the code here to offer a :ship: with my other feedback addressed or not at your discretion.

omarjackman commented:
@dllh I addressed your comments in @cd198cf2803d8e020c05626dd6c011105d4c60a3. I also swapped the teaser with the `isEmailVerified` warning that was there. It shouldn't affect that functionality at all.

omarjackman commented:
cc @jordwest

jordwest commented on client/me/help/main.jsx:
```diff
@@ -73,32 +80,61 @@ const Help = React.createClass( {
 						<h2 className="help__support-link-title">{ this.translate( 'Get in touch' ) }</h2>
-						<p className="help__support-link-content">{ this.translate( 'Can\'t find the answer? Drop us a line and we\'ll lend a hand.' ) }</p>
+						<p className="help__support-link-content">
+							{ this.translate( 'Can\'t find the answer? Drop us a line and we\'ll lend a hand.' ) }
```
How about double-quotes or backticks to avoid escaping the single quotes?

omarjackman replied:
fixed in @b7e0cd6

jordwest commented on client/me/help/main.jsx:
```diff
@@ -73,32 +80,61 @@ const Help = React.createClass( {
+		return (
+			<div className="help__course-teaser">
+				<Card href="#/help/courses">
+					<Gridicon className="help__course-teaser-icon" icon="help" size={ 30 } />
```
This currently shows an eslint error:
> Gridicon size should be one of recommended sizes: 12, 18, 24, 36, 48, 54, 72

omarjackman replied:
I changed it to `36` in @6c9d969

@mrheston Do you have any issues with the look of the button now?

jordwest commented on client/me/help/main.jsx:
```diff
@@ -73,32 +80,61 @@ const Help = React.createClass( {
+				</Card>
+			</div>
+		);
+	},
```
Perhaps we could move this out to it's own component file

omarjackman replied:
This file in general can be restructured a bit so that the functions: `getCoursesTeaser`, `getSupportLinks` and `getHelpfulArticles` are in their own components. I'd save that for another pull request

jordwest commented on client/me/help/main.jsx:
```diff
@@ -108,9 +144,13 @@ const Help = React.createClass( {
+			isEmailVerified,
 		};
 	}
 )( Help );
```
Everything in `ownProps` is automatically passed down to the component without needing to explicitly copy it in the `mapStateToProps`, so it's possible to just use `isCoursesEnabled` directly in the component.

Although I guess `showCoursesTeaser` might not necessarily `===ownProps.isCoursesEnabled` in the future?

omarjackman replied:
> Although I guess showCoursesTeaser might not necessarily === ownProps.isCoursesEnabled in the future?

Thats correct, It'll be composed of another boolean that indicates if the user has the business package.

jordwest commented:
Left a few suggestions, otherwise looks good to me 👍

omarjackman commented:
I'll merge this if @mrheston has no issues with the new size of the grid icon

mrheston commented:
No issues at all! Looks great 👍

mtias commented on client/me/help/style.scss:
```diff
@@ -46,3 +46,38 @@
 	color: $gray-dark;
 }
+
+.help__course-teaser .card {
```
It'd be great to turn this into a prop of the Card component: `Card highlighted` or `Card primary` which would output a class `.card.is-highlighted` or `.card.is-primary`.

omarjackman replied:
@mtias This gets a bit complicated as I have to do something similar but using a different highlight color.
In that case would you do something like `<Card highlighted="orange" />` or just `<Card highlighted/>` and then add css to make it orange.

mtias replied:
@omarjackman what is the case where you need to use a different color? We could do `accent="blue"` etc, or `status` to match notices, but I'd like to see where it is needed first.

omarjackman replied:
@mtias The extra place is here:
![screen shot 2016-08-24 at 1 00 29 pm](https://cloud.githubusercontent.com/assets/1854440/17939959/dd723434-69fa-11e6-9ec8-e7ae5f2d5dd1.png)

mtias replied:
@omarjackman do you mean the yellow? That is the UpgradeNudge component.

mtias replied:
Which already exists as a component.

mtias commented on client/me/help/main.jsx:
```diff
@@ -73,32 +80,61 @@ const Help = React.createClass( {
+	getCoursesTeaser: function() {
+		return (
+			<div className="help__course-teaser">
+				<Card href="#/help/courses">
```
Why the `#` in the url?

mtias commented on client/me/help/main.jsx:
```diff
@@ -73,32 +80,61 @@ const Help = React.createClass( {
+						<span className="help__course-teaser-title">
+							{ this.translate( 'Courses' ) }
+						</span>
+						<span className="help__course-teaser-description" >
```
there's an extra space before the closing `>`.

omarjackman commented:
@mtias I've made note of your card highlighting comment and it is on my list to do in the up coming week.
